### PR TITLE
Update hardware.md

### PR DESCRIPTION
**Pull request by flip111**
https://github.com/rust-embedded/book/issues/140

### CODE REVIEW

File: src/intro/hardware.md [hunk 1]
```diff
@@ -8,37 +8,32 @@ Let's get familiar with the hardware we'll be working with.
...
-- 8 user LEDs arranged in the shape of a compass
+- A second microcontroller: a [STM32F103CBT](https://www.st.com/en/microcontrollers/stm32f103cb.html). This microcontroller is actually part of an on-board programmer and debugger named ST-LINK and is connected to the USB port named "USB ST-LINK".
 
-- A second microcontroller: a STM32F103CBT. This microcontroller is actually part of an on-board
-  programmer and debugger named ST-LINK and is connected to the USB port named "USB ST-LINK".
+For a more detailed features list and further specification of the board take a look at [STMicroelectronics](https://www.st.com/en/evaluation-tools/stm32f3discovery.html) website.
 
-- There's a second USB port, labeled "USB USER" that is connected to the main microcontroller, the
-  STM32F303VCT6, and can be used in applications.
+A word of warning: be careful if you want to apply external signals to the board. The microcontroller STM32F303VCT6 pins take a maximum voltage of 4 volts, but ideally should be driven by the nominal voltage of 3.3 volts. For further information consult the [6.2 Absolute maximum ratings section in the manual](https://www.st.com/resource/en/datasheet/stm32f303vc.pdf)
```
Comment (therealprof) [inline src/intro/hardware.md]: I would not mention "4V" (not even sure whether that's a typo or the maximum tolerable regular 3.3V) but instead say the nominal voltage is 3.3V with only a few pins being 5V tolerant.

File: src/intro/hardware.md [hunk 2]
```diff
@@ -8,37 +8,32 @@ Let's get familiar with the hardware we'll be working with.
...
+- A [magnetometer](https://en.wikipedia.org/wiki/Magnetometer) as part of the [LSM303DLHC](https://www.st.com/en/mems-and-sensors/lsm303dlhc.html) chip.
 
-- A [gyroscope].
+- A [gyroscope](https://en.wikipedia.org/wiki/Gyroscope) as part of the [L3GD20](https://www.pololu.com/file/0J563/L3GD20.pdf) chip.
 
-[gyroscope]: https://en.wikipedia.org/wiki/Gyroscope
+- 8 user LEDs arranged in the shape of a compass.
 
-- 8 user LEDs arranged in the shape of a compass
+- A second microcontroller: a [STM32F103CBT](https://www.st.com/en/microcontrollers/stm32f103cb.html). This microcontroller is actually part of an on-board programmer and debugger named ST-LINK and is connected to the USB port named "USB ST-LINK".
```
Comment (therealprof) [inline src/intro/hardware.md]: While it is possible that there's a new DISC1 version of this board, in many cases the ST-Link/v2 is implemented via a STM32F103C8 MCU. I'd prefer to leave out the the part after the 103 to cater for both cases.

File: src/intro/hardware.md (hunk 1 above)
Comment (flip111) [reply src/intro/hardware.md]: I agree with this.

File: src/intro/hardware.md (hunk 2 above)
Comment (flip111) [reply src/intro/hardware.md]: Not sure what you are getting at. Is there an alternative version of the same board? Shouldn't it have a different part number then?

Comment (therealprof) [reply src/intro/hardware.md]: > Not sure what you are getting at. Is there an alternative version of the same board? Shouldn't it have a different part number then?

Yes. Typically the discoveries have a different part number but in this case there're just three revisions: "On-board ST-LINK/V2 for PCB version A or B or ST-LINK/V2-B for PCB version C and newer". Revision A and B have a STM32F103C8, revision C is a STM32F103CB.

Comment (flip111) [reply src/intro/hardware.md]: What about

> This microcontroller is actually part of an on-board programmer / debugger and is connected to the USB port named "USB ST-LINK".

Comment (therealprof) [reply src/intro/hardware.md]: You mean striking the previous sentence? Works for me.

File: src/intro/hardware.md [hunk 3]
```diff
@@ -8,37 +8,32 @@ Let's get familiar with the hardware we'll be working with.
...
+- A [magnetometer](https://en.wikipedia.org/wiki/Magnetometer) as part of the [LSM303DLHC](https://www.st.com/en/mems-and-sensors/lsm303dlhc.html) chip.
 
-- A [gyroscope].
+- A [gyroscope](https://en.wikipedia.org/wiki/Gyroscope) as part of the [L3GD20](https://www.pololu.com/file/0J563/L3GD20.pdf) chip.
 
-[gyroscope]: https://en.wikipedia.org/wiki/Gyroscope
+- 8 user LEDs arranged in the shape of a compass.
 
-- 8 user LEDs arranged in the shape of a compass
+- A second microcontroller: a [STM32F103CBT](https://www.st.com/en/microcontrollers/stm32f103cb.html). This microcontroller is actually part of an on-board programmer / debugger and is connected to the USB port named "USB ST-LINK".
```
Comment (therealprof) [inline src/intro/hardware.md]: The STM32F103CBT reference is the one that required the fixing...

Comment (flip111) [reply src/intro/hardware.md]: Can you try to edit my patch? is that enabled?

Comment (therealprof) [inline src/intro/hardware.md]:
```suggestion
- A second microcontroller: a [STM32F103](https://www.st.com/en/microcontrollers/stm32f103cb.html). This microcontroller is actually part of an on-board programmer / debugger and is connected to the USB port named "USB ST-LINK".
```

**Pull request merged**